### DISCUSSION TITLE
wagtail.core and wagtail.admin.edit_handlers depreciation

### DISCUSSION THREAD

**Discussion by htabar** [wagtail/wagtail#11664] [Development]
Hey there,

A huge thank you to everyone for your dedication in developing Wagtail.

I've been working on setting up a few Wagtail plugins and projects, but some haven't been updated to accommodate the changes in the old wagtail.core and wagtail.admin.edit_handlers paths! I understand that evolution calls for significant changes, but I wanted to share an idea on how to navigate through these shifts.

I noticed there was a substantial discussion about changing these paths, mentioned here:
https://docs.wagtail.org/en/stable/releases/3.0.html#changes-to-module-paths

By the way, the decision to rename edit_handlers to Panels was a smart move. However, in reality, many plugins and projects on GitHub that previously worked seamlessly with older versions have encountered obstacles. This is especially challenging for newcomers to Wagtail who can't easily perform a simple find and replace!

For these two deprecations, if feasible, I'd even suggest reintroducing the path and the necessary wagtail.core into INSTALLED_APPS. Then, provide the folders and files with a symbolic link to the new path through Python code, if possible.

**ModelAdmin Depreciation:**
I've checked the discussion on [Snippets generalization](https://github.com/wagtail/wagtail/discussions/10740). Regarding the ModelAdmin deprecation, I strongly recommend reconsidering the strategy used for deprecating wagtail.core and wagtail.admin.edit_handlers, as many plugins are built on top of ModelAdmin.

ModelViewSet, SnippetViewSet, and ModelAdmin essentially serve the same purpose. In my opinion, we could eventually have one unified Wagtalish ModelViewSet, with SnippetViewSet inheriting all the methods from ModelViewSet.

My suggestion for ModelAdmin is to avoid following the same path as wagtail.core and wagtail.admin.edit_handlers. Instead, let's avoid leaving ModelAdmin stranded in code with no utility, just to be used by those stalled apps on GitHub!

**Comment by gasman** [comment]
Hi,
I'm not really clear on what you're proposing here. You're saying "don't do the same thing with ModelAdmin that we did with wagtail.core / edit_handlers", but what _should_ we do instead? It seems to me like the only real alternative is "never deprecate anything".
